hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/HadoopConfigurations.java

```diff
@@ -51,6 +53,8 @@ public static org.apache.hadoop.conf.Configuration getHadoopConf(Configuration c
     org.apache.hadoop.conf.Configuration hadoopConf = FlinkClientUtil.getHadoopConf();
     Map<String, String> options = FlinkOptions.getPropertiesWithPrefix(conf.toMap(), HADOOP_PREFIX);
     options.forEach(hadoopConf::set);
+    // Always use flink io factory
+    hadoopConf.set(HoodieStorageConfig.HOODIE_IO_FACTORY_CLASS.key(), HoodieFlinkIOFactory.class.getName());
```

Comment:
`HOODIE_IO_FACTORY_CLASS` is a write option instead of hadoop configuration, should not set it up here.

Reply:
It looks like an internal configuration, why it needs to be explicitly setup.

Reply:
It's an engine specific internal configuration, introduced by #11192 "Introduce IOFactory and a config to set the factory", IIUC, different engine integration implement their own IOFactory, and set the configuration. 
For example,  Hudi-Spark set this configuration in [`DefaultSource`](https://github.com/apache/hudi/blob/8b7bd1391bd61600d23b8defed4cdc7d789502d1/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala#L71), then it'll be used to get correct IOFactory during writing log or base file.